### PR TITLE
stolostron/backlog#21626--enable klusterlet addon

### PR DESCRIPTION
**Pull request by swopebe**

### CODE REVIEW

File: add-ons/submariner/submariner_serv_disc.adoc [hunk 1]
```diff
@@ -2,6 +2,7 @@
 = Managing service discovery for Submariner
 
 The `submariner-addon` component is a *Technology Preview* feature.
+//please use label, see internal doc. we all agreed on/documented how to present tech preview in an arch call a few quarters ago. (if this is still tech preview) --bcs
```
Comment (swopebe) [inline add-ons/submariner/submariner_serv_disc.adoc]: @cadawson need to make sure this is tech preview and if so, please use the syntax that we use across the doc.

File: add-ons/volsync/volsync.adoc [hunk 2]
```diff
@@ -1,4 +1,5 @@
 [#volsync]
 = VolSync persistent volume replication service (Technology Preview)
+//links?
```
Comment (swopebe) [inline add-ons/volsync/volsync.adoc]: This looks like it should be the parent topic @cadawson but no links and there is no other topic...so maybe this needs the links?

File: add-ons/volsync/volsync_convert_backup.adoc [hunk 3]
```diff
@@ -41,7 +41,7 @@ Replace `destination-ns` with the namespace where the persistent volume claim is
 +
 Replace `snapshot_to_replace` with the `VolumeSnapshot` name that you found in the previous step.
 +
-You can update `resources.requests.storage` with a different value, but it is best practice when the value is at least the same size as the initial source persistent volume claim.
+**Best practice:** You can update `resources.requests.storage` with a different value, when the value is at least the same size as the initial source persistent volume claim.
```
Comment (swopebe) [inline add-ons/volsync/volsync_convert_backup.adoc]: This is the correct way to label these.

File: add-ons/klusterlet_managed.adoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
+[#add-ons-klusterlet]
```
Comment (swopebe) [inline add-ons/klusterlet_managed.adoc:1]: @zhiweiyin318 can you review this doc, which is actually part of cluster and not MCE?

You can ignore the rest, which is a peer review.

Also, our template asks for a verification step, so can you please give me a last step to give users?

Comment (zhiweiyin318) [reply add-ons/klusterlet_managed.adoc:1]: looks great to me.  
yes,  the whole steps are on ACM not MCE since there is no `KlusterletAddonConfig` on MCE.

Comment (swopebe) [reply add-ons/klusterlet_managed.adoc:1]: @zhiweiyin318 is there a verification step we can add?

File: add-ons/volsync/volsync.adoc (hunk 2 above)
Comment (swopebe) [reply add-ons/volsync/volsync.adoc]: @cadawson poke

File: add-ons/klusterlet_managed.adoc (hunk 4 above)
Comment (swopebe) [reply add-ons/klusterlet_managed.adoc:1]: @zhiweiyin318 if not verification step, I will leave blank.

File: add-ons/klusterlet_managed.adoc [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+See the following available klusterlet add-ons:
+
+- application-manager
+- cert-policy-controller
+- config-policy-controller
+- iam-policy-controller
+- governance-policy-framework
+- search-collector
+
+The steps to enable the klusterlet add-ons for these managed clusters after the ACM is installed.
```
Comment (cadawson) [inline add-ons/klusterlet_managed.adoc]:
```suggestion
Complete the following steps to enable the klusterlet add-ons for the managed clusters after the {product-title-short} is installed:
```

File: add-ons/klusterlet_managed.adoc [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: true
+----
+
+**Note:** `policyController` controls the `governance-policy-framework` and `config-policy-controller` `managedClusterAddons`.
+
+. Save the file as `klusterlet-addon-config.yaml`.
+. Apply the YAML by running the following command on the hub cluster.
```
Comment (cadawson) [inline add-ons/klusterlet_managed.adoc]:
```suggestion
. Apply the YAML by running the following command on the hub cluster:
```

File: add-ons/submariner/submariner.adoc [hunk 7]
```diff
@@ -91,6 +91,7 @@ install -m744 /tmp/subctl-${VERSION}/subctl-${VERSION}-linux-amd64 /$HOME/.local
 === Using the subctl commands
 
 After adding the utility to your path, the following table briefly describes the available commands:
```
Comment (cadawson) [inline add-ons/submariner/submariner.adoc]:
```suggestion
After adding the utility to your path, view the following table for a brief description of the available commands:
```

Comment (cadawson) [reply add-ons/submariner/submariner.adoc]: Thanks. Good catch!

File: add-ons/volsync/volsync.adoc (hunk 2 above)
Comment (cadawson) [reply add-ons/volsync/volsync.adoc]: Links are added in another PR that is out for review.

File: add-ons/volsync/volsync_replicate.adoc [hunk 8]
```diff
@@ -1,36 +1,40 @@
...
+You can use three methods to replicate persistent volumes with VolSync, which depend on the number of synchronization locations that you have. The Rsync method is used for this example. For information about the other methods and more information about Rsync, see https://volsync.readthedocs.io/en/latest/usage/index.html[Usage] in the VolSync documentation.  
 
-Rsync replication is a one-to-one replication of persistent volumes, and is likely to be the most commonly used. This is used for replicating data to a remote site. 
+Rsync replication is a commonly used, one-to-one replication of persistent volumes. This is used for replicating data to a remote site. 
 
 VolSync does not create its own namespace, so it is in the same namespace as other {ocp-short} all-namespace operators. Any changes that you make to the operator settings for VolSync also affects the other operators in the same namespace. For example, if you change to manual approval for channel updates.
```
Comment (cadawson) [inline add-ons/volsync/volsync_replicate.adoc]:
```suggestion
VolSync does not create its own namespace, so it is in the same namespace as other {ocp-short} all-namespace operators. Any changes that you make to the operator settings for VolSync also affects the other operators in the same namespace, such as if you change to manual approval for channel updates. 
```

File: add-ons/klusterlet_managed.adoc (hunk 4 above)
Comment (zhiweiyin318) [reply add-ons/klusterlet_managed.adoc:1]: yes, we can list the `managedClusterAddon` in the cluster namespace to verify whether the enabled `managedClusterAddons` are created after  the `KlusterletAddonConfig` is created.
```
oc get managedclusteraddons -n <cluster namespace>
```

**Review by cadawson — CHANGES_REQUESTED** [review]
Thanks for the review. Added suggestions.

File: add-ons/klusterlet_managed.adoc (hunk 4 above)
Comment (swopebe) [reply add-ons/klusterlet_managed.adoc:1]: added as verification step

File: add-ons/submariner/submariner_serv_disc.adoc (hunk 1 above)
Comment (cadawson) [reply add-ons/submariner/submariner_serv_disc.adoc]: removed. Thanks.

**Review by cadawson — APPROVED** [review]
Looks great! Thanks for the feedback!

File: add-ons/klusterlet_managed.adoc [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    enabled: true
+  iamPolicyController:
+    enabled: true
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: true
+----
+
+**Note:** `policyController` controls the `governance-policy-framework` and `config-policy-controller` `managedClusterAddons`.
```
Comment (philipwu08) [inline add-ons/klusterlet_managed.adoc:42]: Not sure I understand this note. Maybe the `and` is at the wrong place?

Comment (swopebe) [reply add-ons/klusterlet_managed.adoc:42]: thanks, @zhiweiyin318 does that make sense?

Comment (zhiweiyin318) [reply add-ons/klusterlet_managed.adoc:42]: yes, it's `and`.   `policyController` filed enable/disable the 2 addons .

**Comment by swopebe** [comment]
@zhiweiyin318 can you better explain, not sure I understand what you mean by ^^^? Thanks so much.

**Comment by zhiweiyin318** [comment]
@swopebe  in 2.4 `policyController.enable` only enable/disable `policy-controller` addon. 
but in 2.5 `policy-controller`addon is divided into 2 addons `governance-policy-framework` and `config-policy-controller`, so `policyController` controls the `governance-policy-framework` and `config-policy-controller` `managedClusterAddons`.  there is no `policy-controller` addon in 2.5.

**Comment by swopebe** [comment]
OK, still very confusing and will be confusing to the reader. In 2.5 doc, we don't want to discuss what was previous. I think I have an idea:

The policy-controller`addon is divided into two add-ons: `governance-policy-framework` and `config-policy-controller`. As a result, the `policyController` controls the `governance-policy-framework` and the `config-policy-controller` `managedClusterAddons`.

**Pull request merged**